apple/RNCWebView.m

```diff
@@ -1365,19 +1365,32 @@ - (void)evaluateJS:(NSString *)js
+    // (think phone call, timer, alarms etc), the WkWebView audio will stop playing and will remain silent until the <audio> element is played again.
+    // Looping the audio spawns excessive worker threads causing a higher than normal cpu usage, so instead, the audio is played every 2 seconds.
+    
+    // For more context: https://stackoverflow.com/questions/56460362/how-to-force-wkwebview-to-ignore-hardware-silent-switch-on-ios
     NSString *mp3Str = @"data:audio/mp3;base64,//tAxAAAAAAAAAAAAAAAAAAAAAAASW5mbwAAAA8AAAAFAAAESAAzMzMzMzMzMzMzMzMzMzMzMzMzZmZmZmZmZmZmZmZmZmZmZmZmZmaZmZmZmZmZmZmZmZmZmZmZmZmZmczMzMzMzMzMzMzMzMzMzMzMzMzM//////////////////////////8AAAA5TEFNRTMuMTAwAZYAAAAAAAAAABQ4JAMGQgAAOAAABEhNIZS0AAAAAAD/+0DEAAPH3Yz0AAR8CPqyIEABp6AxjG/4x/XiInE4lfQDFwIIRE+uBgZoW4RL0OLMDFn6E5v+/u5ehf76bu7/6bu5+gAiIQGAABQIUJ0QolFghEn/9PhZQpcUTpXMjo0OGzRCZXyKxoIQzB2KhCtGobpT9TRVj/3Pmfp+f8X7Pu1B04sTnc3s0XhOlXoGVCMNo9X//9/r6a10TZEY5DsxqvO7mO5qFvpFCmKIjhpSItGsUYcRO//7QsQRgEiljQIAgLFJAbIhNBCa+JmorCbOi5q9nVd2dKnusTMQg4MFUlD6DQ4OFijwGAijRMfLbHG4nLVTjydyPlJTj8pfPflf9/5GD950A5e+jsrmNZSjSirjs1R7hnkia8vr//l/7Nb+crvr9Ok5ZJOylUKRxf/P9Zn0j2P4pJYXyKkeuy5wUYtdmOu6uobEtFqhIJViLEKIjGxchGev/L3Y0O3bwrIOszTBAZ7Ih28EUaSOZf/7QsQfg8fpjQIADN0JHbGgQBAZ8T//y//t/7d/2+f5m7MdCeo/9tdkMtGLbt1tqnabRroO1Qfvh20yEbei8nfDXP7btW7f9/uO9tbe5IvHQbLlxpf3DkAk0ojYcv///5/u3/7PTfGjPEPUvt5D6f+/3Lea4lz4tc4TnM/mFPrmalWbboeNiNyeyr+vufttZuvrVrt/WYv3T74JFo8qEDiJqJrmDTs///v99xDku2xG02jjunrICP/7QsQtA8kpkQAAgNMA/7FgQAGnobgfghgqA+uXwWQ3XFmGimSbe2X3ksY//KzK1a2k6cnNWOPJnPWUsYbKqkh8RJzrVf///P///////4vyhLKHLrCb5nIrYIUss4cthigL1lQ1wwNAc6C1pf1TIKRSkt+a//z+yLVcwlXKSqeSuCVQFLng2h4AFAFgTkH+Z/8jTX/zr//zsJV/5f//5UX/0ZNCNCCaf5lTCTRkaEdhNP//n/KUjf/7QsQ5AEhdiwAAjN7I6jGddBCO+WGTQ1mXrYatSAgaykxBTUUzLjEwMKqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqg==";
     NSString *scr;
+
+    int timeIntervalForInaudibleAudioMs = 2000;
     if (initialSetup) {
-        scr = [NSString stringWithFormat:@"var s=new Audio('%@');s.id='wkwebviewAudio';s.controls=false;s.loop=true;s.play();document.body.appendChild(s);true", mp3Str];
+        scr = [NSString stringWithFormat:@"var s=new Audio('%@');s.id='wkwebviewAudio';s.controls=false;s.play();document.body.appendChild(s);var playInterval = setInterval(() => s.play(), %d);true", mp3Str, timeIntervalForInaudibleAudioMs];
     } else {
-        scr = [NSString stringWithFormat:@"document.getElementById('wkwebviewAudio').muted=false", mp3Str];
+        // When the app is foregrounded, the audio element is played again and the interval is restarted.
+        scr = [NSString stringWithFormat:@"var audio=document.getElementById('wkwebviewAudio');audio.muted=false;audio.play();var playInterval = setInterval(() => s.play(), %d);true", timeIntervalForInaudibleAudioMs];
```

Comment:
wouldn't we need to stop the old interval?

Reply:
i see.

and with this 'play on an interval' method, we don't see the same activity as we do when we are looping?

Reply:
the activity is much less
before:
this is  for one of the mediaserverd threads
![Screen Shot 2022-12-20 at 1 37 31 PM](https://user-images.githubusercontent.com/1707258/208741320-fddb110c-b61b-4a46-899e-b7fb58bb467d.png)

after:

![Screen Shot 2022-12-20 at 1 32 26 PM](https://user-images.githubusercontent.com/1707258/208741423-aa9dd51c-ef20-4909-b1c0-2e3d68af8414.png)

i am making the assumption that these threads are equivalent in the sense that both of them are responsible for playing the audio. in the second screenshot, i validated that there is some cpu activity every 2s. i did not see a thread that resembled the first screen shot (which had loop=true)

Reply:
I realized the second screenshot didn't have the timescale for you to verify that it's every 2 s
![Screen Shot 2022-12-20 at 1 32 31 PM](https://user-images.githubusercontent.com/1707258/208742419-848272d4-93f9-4664-854b-0acf64f294fd.png)